praw/objector.py

```diff
@@ -151,13 +155,17 @@ def objectify(
                 return data
             if "things" in data["json"]["data"]:  # Submission.reply
                 return self.objectify(data["json"]["data"]["things"])
+            if "rules" in data["json"]["data"]:
+                return self.objectify(loads(data["json"]["data"]["rules"]))
             if "url" in data["json"]["data"]:  # Subreddit.submit
                 # The URL is the URL to the submission, so it's removed.
                 del data["json"]["data"]["url"]
                 parser = self.parsers[self._reddit.config.kinds["submission"]]
             else:
                 parser = self.parsers["LiveUpdateEvent"]
             return parser.parse(data["json"]["data"], self._reddit)
+        if "rules" in data:
```

Comment:
Why is this recursing rather than just returning the direct result?

And why are there two different returns here pertaining to rules?

Reply:
I'm assuming the first return leads to the second one. And this code was done by jarhill. I just merged it in.

Reply:
`data["rules"]` ends up being a list, so we take advantage of the section of the objector that handles lists.

Reply:
Yeah, the objector code is mine.

Reply:
Wow, this rules stuff is rather interesting. I wonder if it'd make more sense to move this rule-specific parsing to the methods which the calls originate from.

The objectors original intent was to handle the things that are _easy_ to parse. Now it's grown into a massive mess.

Reply:
If you think that would be a good course of action, then sure! That's how @PythonCoderAS did it initially, and a big part of my review comments were about moving over to using the objector.

Reply:
🤷 let's leave it for now. It doesn't hurt (other than my eyes).

Reply:
I mean, it's easy to parse each class separately. It's just that There's no documentation on the sample structure. Maybe each kind of dict could be put in as comments.

Reply:
Adding sample inputs would clutter up the already cluttered code further. The best way to understand (and modify) the objector at the moment is to step through it with a debugger.

Reply:
Maybe we should add a page on the objector in the developer section.